Fraser/1.99.3/fraser.R

```diff
@@ -0,0 +1,158 @@
+
+  message(date(), ": dimension before filtering...")
+  print(dim(assays(fds)$rawCountsJ))
+  print(dim(assays(fds)$rawCountsSS))
+
+  fds <- fds[mcols(fds, type="j")[,"passed"]]
+  message(date(), ": dimension after filtering...")
+  print(dim(assays(fds)$rawCountsJ)) 
+  print(dim(assays(fds)$rawCountsSS))
+
+  #Finds the optimal encoding dimension by injecting artificial splicing outlier ratios while maximizing the precision-recall curve.
+  # Get range for latent space dimension, max tested dimensions = 6
+  mp <- 6 
+  a <- 2
+  b <- min(ncol(fds), nrow(fds)) / mp   # N/mp
+
+  maxSteps <- 12
+  if(mp < 6){
+    maxSteps <- 15
+  }
+
+  Nsteps <- min(maxSteps, b)
+  pars_q <- unique(round(exp(seq(log(a),log(b),length.out = Nsteps))))
+
+  message(date(), ": getEncDimRange", pars_q)
+
+  psiType = "jaccard"
+  impl = "PCA"
+
+  fds <- saveFraserDataSet(fds, name="fds-normfalse")
+  # Heatmap of the sample correlation
+#  plotCountCorHeatmap(fds, type=psiType, logit=FALSE, normalized=FALSE)
+  # Heatmap of the intron/sample expression
+#  plotCountCorHeatmap(fds, type=psiType, logit=FALSE, normalized=FALSE, plotType="junctionSample", topJ=100, minDeltaPsi=0.05)
+
+  #set.seed(as.integer(random))
+  # hyperparameter opimization can be used to estimate the dimension q of the latent space of the data. 
+  # It works by artificially injecting outliers into the data and then comparing the AUC of recalling these outliers for different values of q.
+  fds <- optimHyperParams(fds, type=psiType, implementation=impl, q_param=pars_q, minDeltaPsi=0.05, plot=FALSE)
+  # retrieve the estimated optimal dimension of the latent space
+  #currentType(fds) <- psiType
+  bestq = bestQ(fds, type=psiType)
+  message(date(), "...find best q: " ,bestq)
+
+  # Add verbosity to the FRASER object
+  verbose(fds) <- 3 
+  fds <- fit(fds, q=bestq, type=psiType, implementation=impl, iterations=15)
+
+  # Heatmap of the sample correlation
+#  plotCountCorHeatmap(fds, type=psiType, logit=FALSE, normalized=TRUE)
+  # Heatmap of the intron/sample expression
+#  plotCountCorHeatmap(fds, type=psiType, logit=FALSE, normalized=TRUE, plotType="junctionSample", topJ=100, minDeltaPsi=0.05)
+  dev.off()
+  fds <- saveFraserDataSet(fds, name="fds-normtrue")
+
+  #all in one getEncDimRange, optimHyperParams, bestQ, fit
+  #fds_fraser <- FRASER(fds_filtered, q=c(jaccard=2))
+  
+  fds <- annotateRanges(fds, GRCh = 38)
+
+  # Pvalues
+  fds <- calculatePvalues(fds, type=psiType)
+  # Adjust Pvalues
+  fds <- calculatePadjValues(fds, type=psiType)
+  
+
+  fds <- saveFraserDataSet(fds, name="fds-final")
+  
+  return (fds)
+}
+
+
+write_fraser_output <- function(fraser_ds, prefix){
+  # retrieve results with default and recommended cutoffs (padj <= 0.05 (0.1 gagneurlab) and # |deltaPsi| >= 0.3)
+  fraser_res <- results(fraser_ds, all=TRUE)
+  #res <- results(fds, psiType=psiTypes, aggregate=TRUE, collapse=FALSE, all=TRUE)
+  message(date(), ": result...")
+  print(fraser_res)
```

Comment:
Try to avoid print statements, since they might get lost in the stdout. If you want to save / view the data of a variable, write it to a logfile instead